src/google.js

```diff
@@ -10,13 +10,147 @@
  * governing permissions and limitations under the License.
  */
 const { URL } = require('url');
+const { google } = require('googleapis');
 
 const { fetch } = require('@adobe/helix-fetch').context({
   httpsProtocols:
   /* istanbul ignore next */
   process.env.HELIX_FETCH_FORCE_HTTP1 ? ['http1'] : ['http2', 'http1'],
 });
 const { utils } = require('@adobe/helix-shared');
+const cache = require('./cache');
+
+/**
+ * Remember the access token for future action invocations.
+ */
+let tokenCache = {};
+
+function createOAuthClient(options, creds) {
+  const oAuth2Client = new google.auth.OAuth2(options);
+  oAuth2Client.setCredentials({
+    scope: 'https://www.googleapis.com/auth/drive.readonly',
+    ...tokenCache,
+    ...creds,
+  });
+  oAuth2Client.on('tokens', (tokens) => {
+    tokenCache = tokens;
+  });
+
+  return oAuth2Client;
+}
+
+const createCachedOAuthClient = cache(createOAuthClient);
+
+async function getUncachedIdFromPath(path, parentId, log, options) {
+  const {
+    GOOGLE_DOCS2MD_CLIENT_ID: clientId,
+    GOOGLE_DOCS2MD_CLIENT_SECRET: clientSecret,
+    // eslint-disable-next-line camelcase
+    GOOGLE_DOCS2MD_REFRESH_TOKEN: refresh_token,
+  } = options;
+
+  const auth = await createCachedOAuthClient({ clientId, clientSecret }, { refresh_token });
+
+  const drive = google.drive({
+    version: 'v3',
+    auth,
+  });
+
+  const [name, ...rest] = path.split('/');
+
+  const query = [
+    `'${parentId}' in parents`,
+    `and name = ${JSON.stringify(name)}`,
+    'and trashed=false',
+    // folder if path continues, sheet otherwise
+    `and mimeType = '${rest.length ? 'application/vnd.google-apps.folder' : 'application/vnd.google-apps.spreadsheet'}'`].join(' ');
+  log.debug(query);
+
+  const res = await drive.files.list({
+    q: query,
+    fields: 'files(id, name)',
+    includeItemsFromAllDrives: true,
+    supportsAllDrives: true,
+  });
+
+  const [item] = res && res.data && res.data.files && res.data.files.length
+    ? res.data.files : [];
+
+  if (item && rest.length) {
+    // eslint-disable-next-line no-use-before-define
+    return getIdFromPath(rest.join('/'), item.id, log, options);
+  } else if (item) {
+    return item.id;
+  }
+  return null;
+}
+
+const getIdFromPath = cache(getUncachedIdFromPath);
+
+async function handleJSON(opts) {
+  const {
+    mp, log, options,
+  } = opts;
+
```

Comment:
check if the google secrets are really present. otherwise return a 403 or 500.

Reply:
The Google client checks it already and throws, which leads to a 500